scene/resources/raw_text.cpp

```diff
@@ -0,0 +1,168 @@
+/*************************************************************************/
+/*  raw_text.cpp                                                         */
+/*************************************************************************/
+/*                       This file is part of:                           */
+/*                           GODOT ENGINE                                */
+/*                    http://www.godotengine.org                         */
+/*************************************************************************/
+/* Copyright (c) 2007-2016 Juan Linietsky, Ariel Manzur.                 */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#include "raw_text.h"
+#include "core/object_type_db.h"
+#include "core/os/file_access.h"
+
+void RawText::set_text(const String& text) {
+	if(text.length() != m_text.length() && text!=m_text) {
+		m_text = text;
+		emit_changed();
+	}
+}
+
+void RawText::_bind_methods() {
+
+	ObjectTypeDB::bind_method(_MD("set_text","text"),&RawText::set_text);
+	ObjectTypeDB::bind_method(_MD("get_text"),&RawText::get_text);
+	ADD_PROPERTY( PropertyInfo(Variant::STRING,"text"), _SCS("set_text"), _SCS("get_text"));
+
+}
+
+void RawText::get_recognized_extensions(List<String> *p_extensions) {
+	String extensions[] = {
+		"txt", "md", "json", "toml", "cfg", "ini", "inf", "c", "cpp", "cxx",
+		"c++", "h", "hpp",  "m", "mm", "css", "htm", "html", "py", "java", "js",
+		"sh", "cmd", "csv", "cs", "as", "bash", "hlsl", "glsl", "diff", "php",
```

Comment:
This would override GodotSharp's resource loaded. Or would it be overridden by GodotSharp's?